Model/WebApi/Addresscheck.php

```diff
@@ -0,0 +1,195 @@
+     * @param \Magento\Checkout\Model\Session                       $checkoutSession
+     */
+    public function __construct(
+        \Payone\Core\Model\Risk\Addresscheck $addresscheck,
+        \Payone\Core\Helper\Toolkit $toolkitHelper,
+        \Payone\Core\Model\WebApi\AddresscheckResponseFactory $responseFactory,
+        \Magento\Checkout\Model\Session $checkoutSession
+    ) {
+        $this->addresscheck = $addresscheck;
+        $this->toolkitHelper = $toolkitHelper;
+        $this->responseFactory = $responseFactory;
+        $this->checkoutSession = $checkoutSession;
+    }
+
+    /**
+     * Generate the confirm message from the given address
+     *
+     * @param  \Magento\Quote\Api\Data\AddressInterface $addressData
+     * @return string
+     */
+    protected function getConfirmMessage(\Magento\Quote\Api\Data\AddressInterface $addressData)
+    {
+        $sMessage  = __('Address corrected. Please confirm.')."\n\n";
+        $sMessage .= $addressData->getFirstname().' '.$addressData->getLastname()."\n";
+
+        $mStreet = $addressData->getStreet();
+        if (is_array($mStreet)) { // address can be string
+            $sMessage .= $mStreet[0]."\n"; // add first line of address array
+        } else {
+            $sMessage .= $mStreet."\n"; // add string directly
+        }
+        $sMessage .= $addressData->getPostcode().' '.$addressData->getCity();
+
+        return $sMessage;
+    }
+
+    /**
+     * Return error message
+     *
+     * @return string
+     */
+    protected function getErrorMessage()
+    {
+        $sErrorMessage = $this->toolkitHelper->getConfigParam('stop_checkout_message', 'address_check', 'payone_protect');
+        if (empty($sErrorMessage)) { // add default errormessage if none is configured
+            $sErrorMessage = 'An error occured during the addresscheck.';
+        }
+        return $sErrorMessage;
+    }
+
+    /**
+     * Add the score to the correct session variable
+     *
+     * @param  \Magento\Quote\Api\Data\AddressInterface $oAddress
+     * @param  bool                                     $blIsBillingAddress
+     * @return void
+     */
+    protected function addScoreToSession(\Magento\Quote\Api\Data\AddressInterface $oAddress, $blIsBillingAddress)
+    {
+        $sScore = $this->addresscheck->getScore($oAddress);
+        if ($blIsBillingAddress === true) { // is billing address?
+            $this->checkoutSession->setPayoneBillingAddresscheckScore($sScore);
+        } else {
+            $this->checkoutSession->setPayoneShippingAddresscheckScore($sScore);
+        }
+    }
+
+    /**
+     * Send addresscheck request and handle the response object
+     *
+     * @param  AddresscheckResponse                     $oResponse
+     * @param  \Magento\Quote\Api\Data\AddressInterface $oAddress
+     * @param  bool                                     $blIsBillingAddress
+     * @return AddresscheckResponse
+     */
+    protected function handleAddresscheck(
```

Comment:
This method is too complex, please refactor: https://scrutinizer-ci.com/g/PAYONE-GmbH/magento-2/inspections/f57e816f-b8e5-4e29-b0a2-aa3d5399368d/code-structure/operation/Payone%5CCore%5CModel%5CWebApi%5CAddresscheck%3A%3AhandleAddresscheck